grpcutil/dial.go

```diff
@@ -15,7 +16,9 @@ import (
 // If the URL is prefixed by a protocol (grpc:// or grpcs://) that overrides the TLS flag.
 func Dial(address string, tls bool, caFile, tokenFile string) (*grpc.ClientConn, error) {
 	address, tls = parseAddress(address, tls)
-	return grpc.Dial(address, append(DialOptions(tokenFile), tlsOpt(tls, caFile))...)
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	return grpc.DialContext(ctx, address, append(DialOptions(tokenFile), grpc.WithBlock(), tlsOpt(tls, caFile))...)
```

Comment:
Shouldnt we use [WithReturnConnectionError](https://pkg.go.dev/google.golang.org/grpc#WithReturnConnectionError) instead?

Reply:
Do we mind that WithReturnConnectionError is marked as experimental? If not happy to swap it

Reply:
It just means it can change or be deleted in future releases. As long as we use the same version of the gRPC library, it should not cause any issue.